File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlToken.cs
namespace T1.SqlSharp.Expressions;

public class SqlToken : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.Token;
    public string Value { get; set; } = string.Empty;
    public required TextSpan Span { get; set; }
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_SqlToken(this);
    }

    public string ToSql()
    {
        return Value;
    }

}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlToken.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlToken implements ISqlExpression {
    SqlType: SqlType = SqlType.Token;
    Span: TextSpan = new TextSpan();

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_SqlToken(this);
    }

    ToSql(): string {
        return '';
    }
} 